pallets/broadcast/src/lib.rs

```diff
@@ -61,18 +62,18 @@ pub mod pallet {
 	pub(super) type IncrementalId<T: Config> = StorageValue<_, IncrementalIdType, ValueQuery>;
 
 	#[pallet::storage]
+	#[pallet::whitelist_storage]
 	/// Execution context to figure out where the trade is originated from
 	#[pallet::getter(fn execution_context)]
 	pub(super) type ExecutionContext<T: Config> = StorageValue<_, ExecutionIdStack, ValueQuery>;
 
-	/// To handle the overflow of increasing the execution context.
-	/// After the stack is full, we start to increase the overflow count,
-	/// so we how many times we can ignore the removal from the context.
-	#[pallet::storage]
-	pub(super) type OverflowCount<T: Config> = StorageValue<_, u32, ValueQuery>;
-
 	#[pallet::error]
-	pub enum Error<T> {}
+	pub enum Error<T> {
+		///The execution context has reached its maximum size
+		ExecutionContextFull,
```

Comment:
```suggestion
		ExecutionCallStackOverflow,
```

Reply:
Reasoning : If you get this error you could  infer what you are doing wrong

Reply:
nice one!